tools/logger/convert.c

```diff
@@ -141,6 +141,13 @@ get_uuid_entry(const struct snd_sof_uids_header *uids_dict, uint32_t uid_ptr)
 	       uids_dict->base_address);
 }
 
+uint32_t get_uuid_key(const struct snd_sof_uids_header *uids_dict,
+		      const struct sof_uuid_entry *entry)
+{
+	return (uintptr_t)entry - (uintptr_t)uids_dict -
```

Comment:
What does this do, can you describe how the dictionary format and how the look works.

Reply:
The code was moved into a separate function. This arithmetic is required to translate from the runtime location to an address inside the ldc file. uids_dict->data_offset and uids_dict->base_address are both constants, uids_dict is a mmaped region from the ldc file presumably. We are essentially translating from the unusable address with base 0x1B.. whatever to an actual usable address inside the mmaped .ldc file.

Reply:
base is 0x1FFFA000, but rest is right

Reply:
ok, can we have this in a comment.

Reply:
added